tests/expose_test.go

```diff
@@ -81,34 +94,96 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		Context("Expose ClusterIP service", func() {
 			const servicePort = "27017"
 			const serviceName = "cluster-ip-vmi"
-			It("[test_id:1531][label:masquerade_binding_connectivity]Should expose a Cluster IP service on a VMI and connect to it", func() {
+
+			var vmiExposeArgs []string
+
+			var jobCleanupFunc func() error
+			var serviceCleanupFunc func() error
+
+			BeforeEach(func() {
+				vmiExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachineinstance", "--namespace", tcpVM.GetNamespace(), tcpVM.GetName(),
+					"--port", servicePort, "--name", serviceName,
+					"--target-port", strconv.Itoa(testPort),
+				}
+			})
+
+			BeforeEach(func() {
+				jobCleanupFunc = nil
+			})
+
+			BeforeEach(func() {
+				serviceCleanupFunc = nil
+			})
+
+			AfterEach(func() {
+				Expect(jobCleanupFunc).NotTo(BeNil(), "a successful test must have stored a way to delete the batchv1.Job entity")
+				Expect(jobCleanupFunc()).To(Succeed(), "should be able to delete the batchv1.Job entity")
+			})
+
+			AfterEach(func() {
+				Expect(serviceCleanupFunc).NotTo(BeNil(), "a successful test must have stored a way to delete the k8sv1.Service entity")
+				Expect(serviceCleanupFunc()).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			table.DescribeTable("[test_id:1531][label:masquerade_binding_connectivity]Should expose a Cluster IP service on a VMI and connect to it", func(ipFamily k8sv1.IPFamily) {
 				By("Exposing the service via virtctl command")
-				virtctl := tests.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "virtualmachineinstance", "--namespace",
-					tcpVM.Namespace, tcpVM.Name, "--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort))
-				err := virtctl()
-				Expect(err).ToNot(HaveOccurred())
+				virtctl := tests.NewRepeatableVirtctlCommand(getVirtctlExposeCommand(vmiExposeArgs, ipFamily)...)
+				Expect(virtctl()).To(Succeed(), "should expose a service via `virtctl expose ...`")
 
 				By("Getting back the cluster IP given for the service")
 				svc, err := virtClient.CoreV1().Services(tcpVM.Namespace).Get(serviceName, k8smetav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
+				serviceCleanupFunc = cleanupService(serviceName, tcpVM.GetNamespace())
 				serviceIP := svc.Spec.ClusterIP
 
 				By("Starting a job which tries to reach the VMI via ClusterIP")
 				job := runHelloWorldJob(serviceIP, servicePort, tcpVM.Namespace)
+				jobCleanupFunc = cleanupJob(job.GetName(), job.GetNamespace())
```

Comment:
I would say that as for `serviceCleanupFunc` the same applies for `jobCleanupFunc`, but here the names are not consts or vars so it will be harder. consider moving this to beforeEach() as well, but not a must.